src/core/parameters.cpp

```diff
@@ -485,6 +485,35 @@ namespace gs {
             return params;
         }
 
+        /**
+         * @brief Read optimization parameters from JSON file
+         * @param[in] path Path to the JSON file (can be absolute or relative)
+         * @return Expected OptimizationParameters or error message
+         */
+        std::expected<OptimizationParameters, std::string> read_optim_params_from_custom_json(const std::string path) {
```

Comment:
This is necessary. I recommend to reuse read_optim_params_from_json and add a name if that's the custom file

`read_optim_params_from_json(const std::string& file_name)`

So you can provide the correct filename upon calling this function. Otherwise this is simply duplicated.